librz/include/rz_core.h

```diff
@@ -1053,12 +1053,29 @@ RZ_API bool rz_core_meta_string_add(RzCore *core, ut64 addr, ut64 size, RzStrEnc
 RZ_API bool rz_core_meta_pascal_string_add(RzCore *core, ut64 addr, RzStrEnc encoding, RZ_NULLABLE const char *name);
 
 // cprint.c
+typedef enum {
+	RZ_CORE_PRINT_FORMAT_TYPE_OCTAL = 0,
+	RZ_CORE_PRINT_FORMAT_TYPE_INTEGER,
+	RZ_CORE_PRINT_FORMAT_TYPE_HEXADECIMAL,
+	RZ_CORE_PRINT_FORMAT_TYPE_INVALID,
+} RzCorePrintFormatType;
+
 RZ_API RZ_OWN char *rz_core_print_string_c_cpp(RzCore *core);
 RZ_API RZ_OWN char *rz_core_hex_of_assembly(RzCore *core, const char *assembly);
 RZ_API RZ_OWN char *rz_core_esil_of_assembly(RzCore *core, const char *assembly);
 RZ_API RZ_OWN char *rz_core_assembly_of_hex(RzCore *core, ut8 *hex, int len);
 RZ_API RZ_OWN char *rz_core_esil_of_hex(RzCore *core, ut8 *hex, int len);
 
+RZ_API RZ_OWN char *rz_core_print_hexdump_diff_str(RZ_NONNULL RzCore *core, ut64 aa, ut64 ba, ut64 len);
+RZ_API RZ_OWN char *rz_core_print_dump_str(RZ_NONNULL RzCore *core, RZ_NULLABLE RzCmdStateOutput *state, ut64 addr, ut8 n, int len, RzCorePrintFormatType format);
+RZ_API RZ_OWN char *rz_core_print_hexdump_or_hexdiff_str(RZ_NONNULL RzCore *core, RZ_NULLABLE RzCmdStateOutput *state, ut64 addr, int len);
+RZ_API RZ_OWN char *rz_core_print_hexdump_byline_str(RZ_NONNULL RzCore *core, RZ_NULLABLE RzCmdStateOutput *state, ut64 addr, int len, ut8 size);
+
+#define rz_core_print_hexdump_diff(...)       rz_cons_print_own(rz_core_print_hexdump_diff_str(__VA_ARGS__))
+#define rz_core_print_dump(...)               rz_cons_print_own(rz_core_print_dump_str(__VA_ARGS__))
+#define rz_core_print_hexdump_or_hexdiff(...) rz_cons_print_own(rz_core_print_hexdump_or_hexdiff_str(__VA_ARGS__))
+#define rz_core_print_hexdump_byline(...)     rz_cons_print_own(rz_core_print_hexdump_byline_str(__VA_ARGS__))
+
```

Comment:
These are why the rz_cons_print_own are needed, but I'd rather do something differently rather than the rz_cons_print_own API which is a bit weird. But first, the API above needs to be implemented by respecting the RzCmdStateOutput